lib/segment/src/index/field_index/numeric_index/mod.rs

```diff
@@ -155,14 +155,24 @@ impl<T: Encodable + Numericable> NumericIndex<T> {
         }
     }
 
-    pub fn get_max_values_per_point(&self) -> usize {
+    /// Maximum number of values per point
+    ///
+    /// # Warning
+    ///
+    /// Zero if the index is empty.
+    pub fn max_values_per_point(&self) -> usize {
         match self {
             NumericIndex::Mutable(index) => index.max_values_per_point,
             NumericIndex::Immutable(index) => index.max_values_per_point,
         }
     }
 
     fn range_cardinality(&self, range: &Range) -> CardinalityEstimation {
+        let max_values_per_point = self.max_values_per_point();
+        if max_values_per_point == 0 {
+            return CardinalityEstimation::exact(0);
+        }
```

Comment:
Not 100% sure if we can shortcut here. It would be awesome if someone else could confirm this.

Reply:
looks good like in geo index